api/apps/api/src/modules/access-control/scenarios-acl/scenario-acl.service.ts

```diff
@@ -104,4 +107,45 @@ export class ScenarioAclService implements ScenarioAccessControl {
     });
     return otherOwnersInScenario >= 1;
   }
+
+  async updateUserInProject(
+    scenarioId: string,
+    updateUserInProjectDto: UserRoleInScenarioDto,
+    loggedUserId: string,
+  ): Promise<Either<Permit, void>> {
+    const { userId, roleName } = updateUserInProjectDto;
+    if (!(await this.isOwner(loggedUserId, scenarioId))) {
+      return left(false);
+    }
+
+    const apiDbConnection = getConnection(DbConnections.default);
+    const apiQueryRunner = apiDbConnection.createQueryRunner();
+
+    await apiQueryRunner.connect();
+    await apiQueryRunner.startTransaction();
+
+    try {
+      const existingUserInProject = await this.roles.findOne({
+        where: {
+          scenarioId,
+          userId,
+        },
+      });
+
+      if (!existingUserInProject) {
+        await this.roles.save({
+          scenarioId,
+          userId,
+          roleName,
+        });
+      } else {
+        await this.roles.update({ scenarioId, userId }, { roleName });
+      }
+    } catch (err) {
```

Comment:
It won't work like that. Only the `apiQueryRunner` instance will be transactional, the `this.roles` repo is still "normal" mode.

Use https://docs.nestjs.com/techniques/database (EntityManager) - we also have a few examples of its usage within codebase.,

Reply:
I used the same method as in [`project-acl.service`](https://github.com/Vizzuality/marxan-cloud/blob/develop/api/apps/api/src/modules/access-control/projects-acl/project-acl.service.ts) so I guess both should be changed right?

Reply:
@rubvalave in this case, yes